backend/inspirehep/search/factories/search.py

```diff
@@ -17,6 +17,12 @@
 
 
 def get_search_with_source(search):
+    request_fields = request.values.get("fields", "", type=str)
+    if request_fields:
+        request_fields = request_fields.split(",")
+        request_fields.append("control_number")
+        search = search.source(includes=request_fields)
+        return search
```

Comment:
to be consistent this can be part of the `InspireSearch` already, method like `source_with_requested_fields` and you use it similarly like below.